presto-main/src/test/java/com/facebook/presto/execution/TestCreateTableTask.java

```diff
@@ -30,6 +30,7 @@
 import com.facebook.presto.spi.PrestoException;
```

Comment:
Let's add a unit test similar to `testCreateWithNotNullColumns` that asserts that adding a new table constraint get's reflected in the ConnectorTableMetadata for supported connectors

Reply:
Can you also include some negative tests
- Multiple PRIMARY KEYs
- Duplicate UNIQUE keys for the same named/unnamed constraint

These will be parsed by the parser, but since these are semantically incorrect, we should fail them in the `CreateTableTask`

Reply:
I have left almost all semantic checking to the connector(s) or catalog service. It is impossible to detect all cases of semantic errors and different systems have their own rules for handling these. E.g. some systems require a globally unique name for primary keys, while others do not.
It may be possible to catch some of the common, universal errors early and return, but I'm not sure whether that is worth it. I could be convinced otherwise

Reply:
I think we should have at least the logically impossible cases (such as multiple PKs) covered in presto-main. For others, lets pick a connector (say HMS) and put some negative tests there

Reply:
I've added some additional error checks here that basically validate whether the create table statement is internally consistent. I am reluctant to make these more exhaustive since they would include additional overhead of communicating with the remote catalog